src/sync/update_tx.js

```diff
@@ -97,6 +100,55 @@ async function onSuccessfulTx(tx, db) {
   let txid;
 
   switch (tx.type) {
+    // Approve was accepted. Sending createEvent.
+    case 'APPROVECREATEEVENT': {
+      try {
+        const createTopicTx = await eventFactory.createTopic({
+          oracleAddress: tx.resultSetterAddress,
+          eventName: tx.name,
+          resultNames: tx.options,
+          bettingStartTime: tx.bettingStartTime,
+          bettingEndTime: tx.bettingEndTime,
+          resultSettingStartTime: tx.resultSettingStartTime,
+          resultSettingEndTime: tx.resultSettingEndTime,
+          senderAddress: tx.senderAddress,
+        });
+        txid = createTopicTx.txid;
+      } catch (err) {
+        logger.error(`Error calling EventFactory.createTopic: ${err.message}`);
+        throw err;
+      }
+
+      // Update Topic's approve txid with the createTopic txid
+      const topic = await db.Topics.findOne({ txid: tx.txid });
+      topic.txid = txid;
+      await DBHelper.updateTopicByQuery(db.Topics, topic, { txid: tx.txid });
+
+      // Update Oracle's approve txid with the createTopic txid
+      const oracle = await db.Oracles.findOne({ txid: tx.txid });
```

Comment:
Where are we getting the `oracle` object?

Reply:
oh so then we should change the, `updateTopicByQuery` and `updateOracleByQuery` to not update all those other fields. but if we did that then it can't be used by the sync/index.